src/Type/FileTypeMapper.php

```diff
@@ -130,13 +131,7 @@ private function createResolvedPhpDocBlock(string $phpDocKey, NameScope $nameSco
 	{
 		$phpDocNode = $this->phpDocStringResolver->resolve($phpDocString);
 		if ($this->resolvedPhpDocBlockCacheCount >= 2048) {
-			$this->resolvedPhpDocBlockCache = array_slice(
-				$this->resolvedPhpDocBlockCache,
-				1,
-				null,
-				true,
-			);
-
+			unset($this->resolvedPhpDocBlockCache[array_key_first($this->resolvedPhpDocBlockCache)]);
```

Comment:
As arrays are ordered, removing the first key is quite CPU expensive. The array should be probably reduced by 10-50% at once.

Also I am not exactly sure how only 2048 temporary copied keys can cause dramatic memory peak increase.